Add separate sidebar toggle button

WithoutPants opened the pull request:
Related Discourse post: https://discourse.stashapp.cc/t/query-page-redesign/2064/79?u=withoutpants

- changes the behaviour of the filter icon button on the scene query page to show the filter dialog
- adds a sidebar toggle button to the right sidebar of the scene query page toolbar

<img width="1462" height="70" alt="image" src="https://github.com/user-attachments/assets/97347786-b2a1-40cf-a852-cb1b88fd46f9" />

cj12312021 commented:
Welcome back. I hope the time off went well.

I’ve tested this branch on my development machine and haven’t noticed any bugs. I’ll ingest this into my production machine and get a better feel for how workable the placement of the sidebar toggle button is during regular use.

cj12312021 commented:
I’m not sure how restoring the search input to the toolbar solves the practicality issue of placing the button to access the sidebar on the opposite side of where the sidebar would be open. The location creates unnecessary friction for users, which worsens as the Stash window becomes wider. The UX gives the impression that it doesn’t want users to use the new sidebar, calling into question why effort was invested in implementing it.

I was not aware that the sidebar style is set to hover over list elements at certain viewport sizes. I think, even regardless of the sidebar button placement discussion, that behavior is unideal. Either way, that behaviour would hide a valuable button. If the filter button were kept on the left side, users would no longer be able to access all the filters while the sidebar is open, further incentivizing users not to bother with the sidebar.

Examining how this is handled on other sites, it appears that Bang removes its left sidebar once it has reached this sweet spot and places the sidebar toggle on the right side of the screen, where the button then reveals the sidebar on the right side of the screen. Gamma opts to keep the sidebar in place until the window reaches mobile width, where the sidebar would take up the full screen real estate, with a button placed on the sidebar to toggle it. 

Of the two behaviors, in my opinion, Gamma’s would be the best option, and was actually the behavior I imagined was already in place. Another option could be to place the sidebar toggle and the filter button on the sidebar when this sweetspot is hit, but that might be a lot of work for a window size that is likely niche. Although it sounds like you already had plans to do this for the edit filter button, so including the sidebar toggle there may not be a stretch.

WithoutPants commented:
> I’m not sure how restoring the search input to the toolbar solves the practicality issue of placing the button to access the sidebar on the opposite side of where the sidebar would be open.

Restoring the search input to the toolbar isn't intended to solve the issue of the sidebar toggle location. That's solving a separate issue. 

> The location creates unnecessary friction for users, which worsens as the Stash window becomes wider.

I could make the same argument for moving the filter button on the right side, especially given that the search input is returning to the toolbar. I'm imagining that the sidebar toggle button is going to be less frequently used than the filter button will be, so I'd rather minimise the friction of using the filter button over the sidebar toggle button.

> I was not aware that the sidebar style is set to hover over list elements at certain viewport sizes. I think, even regardless of the sidebar button placement discussion, that behavior is unideal.

On smaller viewports, the sidebar uses far too much horizontal screen real estate to not pop over it. I don't see a huge issue in not being able to access some functions while the sidebar is visible on these viewports since it's not expected that the sidebar will be permanently visible in these instances.

> The UX gives the impression that it doesn’t want users to use the new sidebar, calling into question why effort was invested in implementing it.

I suspect this might be the crux of our disagreements. Based on user feedback and my own usage, when making decisions around the design of this, I don't want to assume that the sidebar will be visible all or even most of the time. As such, the intention of the sidebar shifting more to a supplementary role rather than a primary one. The user needs to be able to perform most functions without the sidebar visible, with the sidebar offering shortcuts to functionality (like filtering on tags/performers for example). 

You may have missed the post describing the design iteration, so I'll repost it here: https://discourse.stashapp.cc/t/query-page-redesign/2064/79?u=withoutpants

WithoutPants commented:
The mouse travel is particularly noticeable on larger screens. I've moved the toggle to the left side on `xl` viewports:

<img width="1890" height="75" alt="image" src="https://github.com/user-attachments/assets/d1f52949-bd8a-4a92-8112-76df8ee6e326" />

I've also adjusted the visibility of the Edit Filter and filter icon buttons based on viewports and whether to sidebar is shown. When the filter icon button is visible, the Edit Filter button is hidden in the sidebar:

<img width="551" height="203" alt="image" src="https://github.com/user-attachments/assets/50f10cc2-81ae-416e-a968-6442660dddbf" />

On smaller screens where the sidebar is visible, the filter icon button is hidden and the Edit Filter button is shown instead to increase the available space for the filter tags:

<img width="1265" height="131" alt="image" src="https://github.com/user-attachments/assets/4c605bff-74de-4b5e-b221-9106789282c0" />

The filter icon button is always visible when the sidebar is hidden.
